client/app/scripts/superdesk-archive/module.js

```diff
@@ -130,6 +130,18 @@ define([
         };
     }
 
+    ArchiveService.$inject = ['desks', 'storage'];
+    function ArchiveService(desks, storage) {
+        this.addTaskToArticle = function (item) {
+            if ((!item.task || !item.task.desk) && desks.activeDeskId && desks.userDesks) {
+                var user = storage.getItem('sess:user');
```

Comment:
there is an api for reading user id via `session` service - it has  `session.identity._id` property